server/middleware/workflow-client.js

```diff
@@ -104,14 +108,42 @@ function uiTransform(item) {
       }
     } else if (Array.isArray(subvalue)) {
       subvalue.forEach(uiTransform);
+    }
+    if (typeof subvalue == 'string') {
+      subvalue = enumTransform(subvalue);
+      item[subkey] = subvalue;
     } else if (subvalue && typeof subvalue === 'object') {
       uiTransform(subvalue);
     }
   });
-
   return item;
 }
 
+function enumTransform(item) {
+  enumPrefixes = [
+    'workflow_execution_status',
+    'event_type_workflow_execution',
+    'event_type_decision_task',
+    'event_type_activity_task',
+    'task_list_kind',
+    'continue_as_new_initiator',
```

Comment:
How did you get this list? Should we list all enums here?

Reply:
i listed the ones that are used in temporal-web.

It's not ideal. As an alternative solution would be making changes to protobuf.js. The decision here is based on the balance between quality, time and our plans. Typically quality and plans are the priorities.
I preferred time here as a) we plan on building new significantly improved web-ui b) use static proto compiler (possibly still protobuf.js in static mode) c) as discussed we are not expecting significant breaking changes after the release.
With the new web-ui the priority is on quality and we can invest in robust handling of enums